pandas/io/json/_normalize.py

```diff
@@ -286,12 +286,15 @@ def _recursive_extract(data, path, seen_meta, level=0):
         else:
             for obj in data:
                 recs = _pull_field(obj, path[0])
-                recs = [
-                    nested_to_record(r, sep=sep, max_level=max_level)
-                    if isinstance(r, dict)
-                    else r
-                    for r in recs
-                ]
+                try:
+                    recs = [
+                        nested_to_record(r, sep=sep, max_level=max_level)
+                        if isinstance(r, dict)
+                        else r
```

Comment:
dont' use large try/excepts, catch in nested_to_record

Reply:
@jreback `recs` can be non iterable. How to catch that in `nested_to_record`?

Reply:
huh? really? you can check if its is_scalar. but does this meet the guarantees?

Reply:
I had to use the `nulls_fixture` which makes it a scalar (see comment above). I rather use `None` which makes it cleaner and more appropriate imho.

Reply:
@bolkedebruin you are missing my point. so it seems that _pull_field can return a scalar and not just an iterable.

can you add a doc-string there (types as well if you can).
then just use is_scalar to detect this

Reply:
@jreback A gotcha. Yes I can do that. Still my comment above stands, I consider using `nulls_fixture` here a bit awkward: the `TypeError` original thrown was due a `None` value. Now I am also testing against a scalar? Is that the right thing? I’ll take you word for it, but I miss some understanding now.

I will update the code to test for `if recs and not is_scalar(xx)`.

Reply:
you reference ``nulls_fixture``, not sure what that has to do with anything. These are values which can be shown as null, they are scalars.

Reply:
For the test I was asked to use the nulls_fixture. This fixture sets the the value in the test to `float(‘NaN’)` which in this case is inherently (in my opinion) different from `None` and changes the nature of the test and the code to pass the test. In other words I do not understand the need for the `nulls_fixture` In this particular case, but maybe I am missing something. Using it does complicate the code and the changes you are requesting from me are related to that increased complexity.

Reply:
@bolkedebruin these are valid values in a dataframe. they must work. i don't think the code is more complex. using a giant try/except is not good practice generally as it tends to suppress things.

Reply:
Both points understood and not argued. I just wonder why this issue wasn’t discovered earlier. I’ll update the PR.

Reply:
> Both points understood and not argued. I just wonder why this issue wasn’t discovered earlier. I’ll update the PR.

no tests that hit the issue. ok thanks.